src/ethRegistrar.ts

```diff
@@ -91,46 +93,20 @@ export function handleNameRenewed(event: NameRenewedEvent): void {
 
 export function handleNameTransferred(event: TransferEvent): void {
   let label = uint256ToByteArray(event.params.tokenId)
-  let registration = new Registration(label.toHex())
-  registration.registrant = event.params.to.toHex()
-  registration.save()
-
   let transferEvent = new NameTransferred(createEventID(event))
-  transferEvent.registration = registration.id
+  transferEvent.registration = label.toHex()
   transferEvent.blockNumber = event.block.number.toI32()
   transferEvent.transactionID = event.transaction.hash
-  transferEvent.newOwner = registration.registrant
-  transferEvent.save()
-}
-
-// Helper for concatenating two byte arrays
-function concat(a: ByteArray, b: ByteArray): ByteArray {
-  let out = new Uint8Array(a.length + b.length)
-  for (let i = 0; i < a.length; i++) {
-    out[i] = a[i]
-  }
-  for (let j = 0; j < b.length; j++) {
-    out[a.length + j] = b[j]
+  // This registration is loaded to refer to registration.registrant.
+  let registration = Registration.load(label.toHex())
+  // This is when minting new name, called by _mint()
```

Comment:
Why would this happen? The only way to register a new name is to call `register` on the registrar contract, and that outputs a `NameRegistered` event.

Reply:
I've already explained as `registrar triggers _mint event which triggers handleNameTransferred where it was failing to save registration schema.`